### ISSUE TITLE
http://staging.chembiohub.ox.ac.uk/all_deps2.box missing

### ISSUE THREAD

**Issue by ningyu18** [thesgc/chembiohub_ws#3]
I got the following error while running "vagrant up":

Bringing machine 'default' up with 'virtualbox' provider...
==> default: Box 'http://staging.chembiohub.ox.ac.uk/all_deps2.box' could not be found. Attempting to find and install...
    default: Box Provider: virtualbox
    default: Box Version: >= 0
==> default: Box file was not detected as metadata. Adding it directly...
==> default: Adding box 'http://staging.chembiohub.ox.ac.uk/all_deps2.box' (v0) for provider: virtualbox
    default: Downloading: http://staging.chembiohub.ox.ac.uk/all_deps2.box
An error occurred while downloading the remote file. The error
message, if any, is reproduced below. Please fix this error and try
again.

The requested URL returned error: 404 Not Found

Has all_deps2.box been moved or renamed?

**Comment by strets123** [comment]
Hi, thank you for alerting us to this issue and sorry you experienced it. The box needs re creating for our latest version. It is now high on our priority list and we will have a fix and update you by end of tomorrow.

**Comment by strets123** [comment]
Hi, there is now a new box url and an up to date box with an example user and an example chemical registration project. Instructions for using the box have also been updated.

Thank you for your patience and do let us know if you have any more issues and what your experience with the product is like.

There are now some basic install docs for ubuntu server that will give you a production install, the current vagrant box is for development but watch this space for new boxes and simplified installation instructions.

**Comment by ningyu18** [comment]
​Thanks very much for attending to this issue. I now have the box up and
running, and was able to run the ​npm and bower commands. However, I
encountered the following error on the next step:

(dev)vagrant@vagrant-ubuntu-trusty-64:~/chembiohub_ws$ python manage.py
migrate
No Secret settings, using default secret key which is insecure
Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File
"/home/vagrant/anaconda2/envs/dev/lib/python2.7/site-packages/django/core/management/__init__.py",
line 385, in execute_from_command_line
    utility.execute()
  File
"/home/vagrant/anaconda2/envs/dev/lib/python2.7/site-packages/django/core/management/__init__.py",
line 354, in execute
    django.setup()
  File
"/home/vagrant/anaconda2/envs/dev/lib/python2.7/site-packages/django/__init__.py",
line 21, in setup
    apps.populate(settings.INSTALLED_APPS)
  File
"/home/vagrant/anaconda2/envs/dev/lib/python2.7/site-packages/django/apps/registry.py",
line 85, in populate
    app_config = AppConfig.create(entry)
  File
"/home/vagrant/anaconda2/envs/dev/lib/python2.7/site-packages/django/apps/config.py",
line 87, in create
    module = import_module(entry)
  File
"/home/vagrant/anaconda2/envs/dev/lib/python2.7/importlib/__init__.py",
line 37, in import_module
    __import__(name)
ImportError: No module named chembl_core_db
(dev)vagrant@vagrant-ubuntu-trusty-64:~/chembiohub_ws$

Any idea why that is?

Thanks again!

Ning

**Comment by strets123** [comment]
Hi Ning,

Did you see the --recursive where it asked you to clone?

git clone  --recursive  git@github.com:thesgc/chembiohub_ws.git

If that was not the issue try

git submodule init 
git submodule update
git submodule foreach git pull origin master

You should see that the directory structure looks like this if you run:

      tree -l -L 1

chembiohub_ws$ tree -l -L 1
.
├── anaconda_requirements.txt
├── cbh_chembl_id_generator -> ./src/cbh_chembl_id_generator/cbh_chembl_id_generator/
├── cbh_chembl_model_extension -> ./src/cbh_chembl_model_extension/cbh_chembl_model_extension/
├── cbh_chembl_ws_extension -> ./src/cbh_chembl_ws_extension/cbh_chembl_ws_extension/
├── cbh_core_model -> ./src/cbh_core_model/cbh_core_model/
├── cbh_core_ws -> ./src/cbh_core_ws/cbh_core_ws/
├── cbh_datastore_model -> ./src/cbh_datastore_model/cbh_datastore_model/
├── cbh_datastore_ws -> ./src/cbh_datastore_ws/cbh_datastore_ws/
├── chembl_business_model -> ./src/chembl_business_model/chembl_business_model/
├── chembl_core_db -> ./src/chembl_core_db/chembl_core_db/
├── chembl_core_model -> ./src/chembl_core_model/chembl_core_model/
├── chembl_extras -> ./src/chembl_extras/chembl_extras/
├── deployment
├── echo_supervisor_config.sh
├── example_custom_deployment
├── flowjs -> ./src/django-flowjs/flowjs/
├── generate_port.py
├── generate_secret_key.py
├── __init__.py
├── install_centos.md
├── install_linux64.sh
├── install_ubuntu.md
├── LICENSE
├── manage.py
├── migrating_data.md
├── pip_requirements.txt
├── README.md
├── scripts
├── src
└── Vagrantfile

**Comment by strets123** [comment]
The chembl_core_model is one of the folders with a symlink. I have not tried this on a windows machine, it is possible the symlinks don't work.

**Comment by ningyu18** [comment]
You are right it was the symbolic links that were causing the trouble on my
Windows machine. I was able to get the code up and running on my Mac using
the latest instructions, although I had to add "sudo" to "npm install" to
get grunt to work correctly. The app looks really impressive, nothing like
the dated and uninspiring corporate registration systems I have seen. I
will play with it a bit more in the next few days. Thanks for the help!

Ning

**Comment by strets123** [comment]
Thanks Ning, it's great to hear that you like the tool, we will fix the windows issue soon and also release other more production-ready installations and documentation.

If you would like to contact me on andrewe.stretton@chembiohub.ox.ac.uk I would be happy to work with you to get the tool configured for your needs.

To configure the fields for yourself there are 2 areas in the admin interface (menu link top right (open in new tab)) to be aware of:

Custom Field Configs allow you to set up a set of fields for the metadata on a compound batch.

These are linked to projects.

Project Types are also linked to projects and allow you to configure whether compounds are displayed or not.

If you have some form of nested experimental data to work with this can be managed by AssayReg.

Best wishes and good luck using the tool.

Andy

**Comment by ningyu18** [comment]
Thanks. I was just curious about what your system looks like and am all set
for now. I am supporting a corporate compound registration system that has
been in use for many years at a research site in a medium/large pharma
company. Our company is now looking to build a global system for all our
research sites, but the current strategy is to take the commercial route.

In any case, thanks again and keep up the good work!

Ning
